client/src/views/files/handler/editor/FileEditor.vue

```diff
@@ -140,38 +218,61 @@ async function openFileWithCryptpad(): Promise<void> {
   try {
     await cryptpadApi.open(cryptpadConfig);
     window.electronAPI.log('info', 'CryptPad editor initialized successfully');
-  } catch (error) {
-    throwError(`Failed to open CryptPad editor: ${error}`);
+  } catch (e: unknown) {
+    if (e instanceof CryptpadError) {
+      let title: Translatable = 'fileViewers.errors.titles.genericError';
+      let message: Translatable = 'fileViewers.errors.informationEditDownload';
+      const level: InformationLevel = InformationLevel.Info;
+
+      switch (e.code) {
+        case CryptpadErrorCode.NotInitialized:
+          title = 'fileViewers.errors.titles.genericError';
+          message = 'fileViewers.errors.informationEditDownload';
+          break;
+        case CryptpadErrorCode.DocumentTypeNotEnabled:
+          title = 'fileViewers.errors.titles.editionNotAvailable';
+          message = 'fileViewers.errors.informationEditDownload';
+      }
+
+      await informationManager.present(
+        new Information({
+          title,
+          message,
+          level,
+        }),
+        PresentationMode.Modal,
+      );
+    }
   }
 }
 
 function validateContentInfo(contentInfo: FileContentInfo | undefined): void {
   if (!contentInfo) {
-    throw new Error('Content info is not available');
-  }
-
-  if (!contentInfo.extension || contentInfo.extension.trim() === '') {
-    throw new Error('File extension is missing or empty');
-  }
+    error.value = 'fileEditors.errors.contentInfoMissing';
```

Comment:
You do not need to throw, but you do need to return, otherwise all other conditions are going to be checked and the last one will win.